### PR TITLE
React Navigation (rebased PR)

### PR DESCRIPTION
**Pull request by FruitieX**
https://reactnavigation.org/ :rocket:

### CODE REVIEW

**Comment by FruitieX** [comment]
TODO: Tab bar icons on iOS

**Comment by krivachy** [comment]
Code looks good I think there are 3 changes that need to be made before merging:

1) JS tests are failing.
2) Icons for tab bar :)
3) Behaviour of the app/navigator has changed and the look and feel of the app has changed between android/iOS. The stack navigator now overlays the tabbar, previously it was in the same tab. Personally I think the use-case when the screens stack inside the tab makes more sense (just look at Twitter/instagram/etc.) Also if we're going for Android Material UI it might make sense to go full on.

output

simulator screen shot 6 mar 2017 23 47 38

simulator screen shot 6 mar 2017 23 47 43

**Comment by FruitieX** [comment]
Thanks for the review.

1. The testing issue might be solved by a RN version upgrade, but in the mean time there's a workaround which I will commit.
2. Will do :)
3. I think I first had the navigators the other way around, which resulted in the AppBar being below the Tab bar which looked a little funny. :) But let's see if it's possible to remedy somehow.

About material-ui, I have something that looks like this at the moment:

screenshot_20170307-114954

**Comment by FruitieX** [comment]
OK, tests are fixed and iOS now looks like this:

simulator screen shot 07 mar 2017 13 48 59
simulator screen shot 07 mar 2017 13 48 52

**Comment by krivachy** [comment]
Header seems to jump on first render and when switching screens on iOS. This is with Debugging enabled, but the jump is still visible without debugging enabled:

[Screen gif 1](https://cloud.githubusercontent.com/assets/5367669/23728372/5ff506cc-0454-11e7-9c74-8781c2eec519.gif)
[Screen gif 2](https://cloud.githubusercontent.com/assets/5367669/23728375/63b70332-0454-11e7-9302-cbd6093c2bf5.gif)

**Comment by FruitieX** [comment]
I've been unable to reproduce the header title jumping around in the iOS simulator.

I'd say it's an issue with [react-navigation](https://github.com/react-community/react-navigation). There was a [somewhat related issue](https://github.com/react-community/react-navigation/pull/562) recently closed, maybe the problem is fixed now.

File: src/modules/navigator/NavigatorState.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {fromJS} from 'immutable';
+
+import AppNavigator from './Navigator';
+
+export default function NavigatorReducer(state, action) {
+  const newState = AppNavigator.router.getStateForAction(action, state && state.toJS());
+  return fromJS(newState || {});
```
Comment (FruitieX) [inline src/modules/navigator/NavigatorState.js]: This is very, very bad.

**Comment by sercanov** [comment]
Hey @FruitieX ,
any ETA on this ? :)

**Comment by FruitieX** [comment]
@sercanov I'll fix the navigation reducer today and then hopefully this PR can be merged

**Comment by FruitieX** [comment]
Awaiting review, then we are good to go :+1:

**Comment by sercanov** [comment]
I'll definitely will try and give feedback asap. I'm sick of NavigationExperimental enough.. 

Actually I integrated react-navigation into pepperoni in my local, its more stable than NE but it still has lot to go, i think they'll release stable v1 in april so we should follow [here](https://github.com/react-community/react-navigation/issues/723) to update accordingly.

Cheers and thanks!

File: src/modules/colors/ColorView.js [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
   static displayName = 'ColorView';
 
+  static navigationOptions = {
+    title: 'Colors!',
+    tabBar: () => ({
+      icon: (props) => (
+        <Icon name='color-lens' size={24} color={props.tintColor} />
+      )
+    }),
+    // TODO: move this into global config?
```
Comment (tino-junge) [inline src/modules/colors/ColorView.js]: Do you still want to move the header config somewhere else? Otherwise let's remove the TODO ;)

**Comment by tino-junge** [comment]
Reviewed once more and as discussed we keep the navigation and icon behavior for android for now and change it if need be. Good job @FruitieX !

**Pull request merged**